From 32dc9c6625e79903364271df66c668cfe12ff8cb Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Mon, 19 Jul 2021 20:14:04 +0300
Subject: [PATCH] [SW-44448] gaudi2: add support for gaudi2 dynamic register

This patch adds support for defining dynamic register mapping to be passed
to lkd for various communication. This is required for dynamic image
load support for gaudi2 in preboot from lkd.

Change-Id: I2cccae0a3c5c4a3221041a05743026ba03eeb21c
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 12b111a..16fd624 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -5,4 +5,49 @@
  *
  */
 
-/* TODO: Commons functions in arc, arm to be moved here. */
+#include <stdint.h>
+#include <preboot.h>
+#include <cpu_if_regs.h>
+#include <cmd_module.h>
+#include <psoc_global_conf_regs.h>
+#include <gaudi2/gaudi2_reg_map.h>
+
+void lkd_dev_spec_regs(struct cpu_dyn_regs *regs)
+{
+	regs->cpu_pq_base_addr_low = mmCPU_IF_PQ_BASE_ADDR_LOW;
+	regs->cpu_pq_base_addr_high = mmCPU_IF_PQ_BASE_ADDR_HIGH;
+	regs->cpu_pq_length = mmCPU_IF_PQ_LENGTH;
+	regs->cpu_pq_init_status = mmCPU_IF_QUEUE_INIT;
+	regs->cpu_eq_base_addr_low = mmCPU_IF_EQ_BASE_ADDR_LOW;
+	regs->cpu_eq_base_addr_high = mmCPU_IF_EQ_BASE_ADDR_HIGH;
+	regs->cpu_eq_length = mmCPU_IF_EQ_LENGTH;
+	regs->cpu_eq_ci = mmCPU_IF_EQ_RD_OFFS;
+	regs->cpu_cq_base_addr_low = mmCPU_IF_CQ_BASE_ADDR_LOW;
+	regs->cpu_cq_base_addr_high = mmCPU_IF_CQ_BASE_ADDR_HIGH;
+	regs->cpu_cq_length = mmCPU_IF_CQ_LENGTH;
+	regs->cpu_pf_pq_pi = mmCPU_IF_PF_PQ_PI;
+	regs->cpu_boot_dev_sts0 = mmCPU_BOOT_DEV_STS0;
+	regs->cpu_boot_dev_sts1 = mmCPU_BOOT_DEV_STS1;
+	regs->cpu_boot_err0 = mmCPU_BOOT_ERR0;
+	regs->cpu_boot_err1 = mmCPU_BOOT_ERR1;
+	regs->cpu_boot_status = mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
+	regs->fw_upd_sts = mmUPD_STS;
+	regs->fw_upd_cmd = mmUPD_CMD;
+	regs->fw_upd_pending_sts = mmUPD_PENDING_STS;
+	regs->fuse_ver_offset = mmFUSE_VER_OFFSET;
+	regs->preboot_ver_offset = mmPREBOOT_VER_OFFSET;
+	regs->uboot_ver_offset = mmUBOOT_VER_OFFSET;
+	regs->hw_state = mmHW_STATE;
+	regs->kmd_msg_to_cpu = mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU;
+	regs->cpu_cmd_status_to_host = mmCPU_CMD_STATUS_TO_HOST;
+	regs->gic_tpc_qm_irq_ctrl = mmGIC_TPC_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_mme_qm_irq_ctrl = mmGIC_MME_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_dma_qm_irq_ctrl = mmGIC_DMA_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_rot_qm_irq_ctrl = mmGIC_ROT_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_nic_qm_irq_ctrl = mmGIC_NIC_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_dma_core_irq_ctrl = mmGIC_DMA_CR_IRQ_CTRL_POLL_REG;
+	regs->gic_host_pi_upd_irq = mmGIC_HOST_PI_UPD_IRQ_POLL_REG;
+	regs->gic_host_halt_irq = mmGIC_HOST_HALT_IRQ_POLL_REG;
+	regs->gic_host_ints_irq  = mmGIC_HOST_INTS_IRQ_POLL_REG;
+	regs->gic_host_soft_rst_irq = mmGIC_HOST_SOFT_RST_IRQ_POLL_REG;
+}
